pkg/scheduler/framework/types.go

```diff
@@ -96,6 +96,7 @@ type FitError struct {
 	Pod                   *v1.Pod
 	NumAllNodes           int
 	FilteredNodesStatuses NodeToStatusMap
+	FailedPlugins         sets.String
```

Comment:
as mentioned above, lets add PluginName to Status.

Reply:
+1

Reply:
I tried but gave up this idea. Some reason explained in https://github.com/kubernetes/kubernetes/pull/98041#discussion_r561618149.

Another reason is: we're now using `Status` as the output of each extension point, as well as individual plugin. For most of the cases, we can directly use the `Status` returned from plugin as the result of extension point. However, for Filtering it's not the case - we merged N results into one:
 https://github.com/kubernetes/kubernetes/blob/44f8fcfe9482ecd6da7e9e88e0e25387ff2ef7fd/pkg/scheduler/framework/runtime/framework.go#L645

That means if we want to use a single `Status` to fit all cases, we have to embed a `[]Failure` to `Status`, and `Failure` to be a tuple with "reason" and "plugin". So a `Status` can be of 3 different styles:

- `len(failures) == 1` - result of a plugin, also act as result of non-FIlter extension point
- `len(failures) == N` - result of Fitler extension oint
- `len(failures) == 0` - abnormal error

I'm not a fan of this as it hides a lot of implicities inside the code.

And, even we go with this design, we still need to inject that status to FitError (or change the signature of sched.recordSchedulingFailure to accpect a Status).

So after considering all the things mentioned, I think the current approach is cleaner and results in way fewer code changes in both core and downstream. WDYT?